### PR TITLE
62001_subject_task_management_for_trainee

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Trainee task management for subjects

### CODE REVIEW

File: app/controllers/user_tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class UserTasksController < ApplicationController
```
Comment (dieunb) [inline app/controllers/user_tasks_controller.rb]: we no need this controller

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
 
   def index
     @user_courses = UserCourse.course_for_cuuent_user current_user.id
   end
+
+  private
+  def get_user_courses_and_tasks
+    @user_courses = @user.user_courses
+    @active_user_course = @user_courses.find_by course_id: Course.active.first.id
+    @user_tasks = UserTask.where("user_id = ?", params[:id])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: move where to scope in model

File: app/controllers/user_subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class UserSubjectsController < ApplicationController
+  def update
+    user_subject = UserSubject.find(params[:id])
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: remove `()`

File: app/models/user_course.rb [hunk 4]
```diff
@@ -6,6 +6,7 @@ class UserCourse < ActiveRecord::Base
   has_many :user_subjects, dependent: :destroy
 
   scope :course_for_current_user, -> (user_id) {where user_id: user_id}
+  scope :current_active_course, -> {UserCourse.find_by course_id: Course.active.first.id}
```
Comment (dieunb) [inline app/models/user_course.rb]: this scope will hit to database 2 twice

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,7 +1,15 @@
...
 
   def index
     @user_courses = UserCourse.course_for_cuuent_user current_user.id
   end
+
+  private
+  def get_user_courses_and_tasks
+    @user_courses = @user.user_courses
+    @active_user_course = @user_courses.current_active_course Course.active.first.id
+    @user_tasks = UserTask.task_for_current_user params[:id]
```
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: scope name ``task_for_current_user`` then we can use ``current_user`` as parameter

**Pull request merged**